src/platform/lumin-runtime/elements/properties/text-children-property.js

```diff
@@ -3,32 +3,32 @@
 import { PropertyDescriptor } from './property-descriptor.js';
 
 export class TextChildrenProperty extends PropertyDescriptor {
-    constructor(name, setterName, isNativeSetter) {
-        super(name, setterName, isNativeSetter)
-    }
+  validate (value) {
+    this.throwIfNotText(value);
+    return true;
+  }
 
-    validate(value) {
-        this.throwIfNotText(value);
-        return true;
-    }
-    
-    throwIfNotText(children) {
-        TextChildrenProperty.throwIfNotText(children);
-    }
+  throwIfNotText (children) {
+    TextChildrenProperty.throwIfNotText(children);
+  }
+
+  static _validateChildren (children) {
+    let valid = true;
+
+    valid &= Array.isArray(children)
+      ? children.every(child => this._validateChildren(child))
```

Comment:
Should we break out a separate function for validating a single child? As is, this would allow children to be an array of arrays of text (or deeper, to arbitrary depth of nested arrays).

Reply:
Having array of strings (or array of an array of strings) is not a practice to specify the text content for HTML style UI declaration. In case of React, we need to process array because the following possible content for inner-text:
1. `<Text>message</Text>` will result in `message` as single string content
2  `<Text>123</Text>` will result in `123` as single number content
3  `<Text>123 messages</Text>` will result in array: `[123, 'messages']` - this is why we need to process an array.

Reply:
Thanks for the explanation on how we end up with arrays (I wasn't clear on that).

What I was referring to, though, is how the _validateChildren function is used recursively, so something like `[123, 'messages', ['nested', 'array', ['we', 'must', 'go', 'deeper!']]]` would also be considered valid.

Reply:
I agree, however React won't produce such `children` value. At least I don't expect :)

Reply:
Unless, the developer intentionally does something like: 
<Text>{ [123, 'messages', ['nested', 'array', ['we', 'must', 'go', 'deeper!']]] }</Text>